TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -389,6 +389,49 @@ public override bool Visit(DataDefinition dataDefinition)
 
             }
 
+            //Check if variable is Sync inside Strict Type definition
+            if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.InsideTypedef) &&
+                dataDefinition.SemanticData.HasFlag(Symbol.Flags.Sync))
+            {
+                //Typedef instruction => check if it's marked Strict
+                if (dataDefinition.SemanticData.Kind == Symbol.Kinds.Typedef && dataDefinition.SemanticData.HasFlag(Symbol.Flags.Strict))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"Cannot declare Type definition {dataDefinition.Name} with Sync clause because it is Strict.");
+                }
+                //Variable inside Typedef => check if parent typedef is marked Strict
+                else if (dataDefinition.SemanticData.NearestParent(Symbol.Kinds.Typedef).HasFlag(Symbol.Flags.Strict))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"{dataDefinition.Name} cannot be declared with Sync clause because it is part of Strict Type definition {dataDefinition.ParentTypeDefinition?.Name}.");
+                }
+
+            }
+            
+            //Variable is of a user defined Strict Type
+            if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.HasATypedefType) && dataDefinition.DataType.RestrictionLevel == RestrictionLevel.STRICT)
+            {
+                //Sync clause not allowed on variable declaration because its type is Strict
+                if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.Sync))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"{dataDefinition.Name} cannot be declared with Sync clause because its Type definition {dataDefinition.DataType.Name} is Strict.");
+                }
+                //Check whether variable is under a sync parent
+                else if (dataDefinition.CodeElement?.LevelNumber?.Value > 1)
+                {
+                    //Retrieve higher parent
+                    Symbol parent = dataDefinition.SemanticData.Owner;
```

Comment:
I thought that sync flag were inherited.
Anyway, can you replace this code by an inheritance of Sync flag?
This way the `if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.Sync))` will be enough to test this rule.

Reply:
Could you update your branch with latest commits from 1721_SemanticDomainLite please ? 558b8e75295dd0dab6d8c8fcd330bdc4e8b194f3 should give you inherited SYNC flags.

Reply:
Ok now sync flag is inherited, but the information that it is inherited is not present. So those two cases are indistinguishable using only Symbol class:
```cob
01 MyGroup sync.
    05 MyTypeVar type MyTypeStrict.
```
and
```cob
01 MyGroup.
    05 MyTypeVar type MyTypeStrict sync.
```